core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -513,3 +513,81 @@ class CollectionAccumulator[T] extends AccumulatorV2[T, java.util.List[T]] {
     getOrCreate.addAll(newValue)
   }
 }
+
+
+/**
+ * An [[AccumulatorV2 counter]] for collecting a list of (mapper id, row count).
+ *
+ * @since 3.4.0
+ */
+class MapperRowCounter extends AccumulatorV2[jl.Long, java.util.List[java.util.List[jl.Long]]] {
+
+  private var _agg: java.util.List[java.util.List[jl.Long]] = _
```

Comment:
why it's a list of list? we need row counts for N mappers, and a list of integer should work.

Reply:
Which of the following that you are thinking:
1. a list of integers where every two forms a pair for partition id and its row count?
2. a list of integers that the index is the mapper/partition id and the value is the row count?

Reply:
a list of tuple2[int, int]?

Reply:
Changed to use Scala tuple